assets/src/components/with-meta-block-edit.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	InspectorControls,
+	withColors,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { Fragment } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { dateI18n, __experimentalGetSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import { ResizableBox, ColorSettings, TextSettings } from './';
+import { maybeEnqueueFontStyle } from '../helpers';
+
+// @todo: Use minimal <RichText> when props.isEditable is true.
+// @todo: Allow individual blocks to add custom controls.
+const MetaBlockEdit = ( props ) => {
+	const {
+		blockContent,
+		attributes,
+		setAttributes,
+		className,
+		fontSize,
+		isSelected,
+		setFontSize,
+		backgroundColor,
+		textColor,
+		setBackgroundColor,
+		setTextColor,
+		toggleSelection,
+		tagName,
+	} = props;
+
+	const {
+		align,
+		ampFontFamily,
+		customFontSize,
+		height,
+		width,
+		opacity,
+	} = attributes;
+
+	const minTextHeight = 20;
+	const minTextWidth = 30;
+	const userFontSize = fontSize.size ? fontSize.size + 'px' : '2rem';
+
+	const ContentTag = tagName;
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( value ) => setAttributes( { align: value } ) }
+				/>
+			</BlockControls>
+			<InspectorControls>
+				<TextSettings
+					fontFamily={ ampFontFamily }
+					setFontFamily={ ( value ) => {
+						maybeEnqueueFontStyle( value );
+						setAttributes( { ampFontFamily: value } );
+					} }
+					fontSize={ fontSize }
+					setFontSize={ setFontSize }
+					customFontSize={ customFontSize }
+				/>
+				<ColorSettings
+					backgroundColor={ backgroundColor }
+					setBackgroundColor={ setBackgroundColor }
+					textColor={ textColor }
+					setTextColor={ setTextColor }
+					fontSize={ fontSize }
+					opacity={ opacity }
```

Comment:
Opacity doesn't seem to be working on the editor side.

Reply:
Ugh, indeed.

I am getting `A component is changing an uncontrolled input of type number to be controlled` erros in the console. Probably related to another issue I just saw with the styling, caused by the components now being nested within an extra div:

![Screenshot 2019-04-10 at 10 56 35](https://user-images.githubusercontent.com/841956/55865862-7e553b80-5b7f-11e9-81e8-c693140d722d.png)

Interestingly there's no console error for the text block, but the styling is off there as well.

Reply:
Fixed now with the latest refactoring